### PR TITLE
BUG: Join did not work correctly when one MultiIndex had only one level

### LINKED ISSUE #36909: BUG: Left join is broken with a MultiIndex with only one level
- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest version of pandas.
- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import pandas as pd

foo = pd.DataFrame(data={'c': 3}, index=pd.MultiIndex.from_tuples([(1, 2)], names=('a', 'b')))
bar = pd.DataFrame(data={'d': 4}, index=pd.MultiIndex.from_tuples([(2,)], names=('b',)))
joined = foo.join(bar, how='left')
print(joined)
```

#### Expected Output

```
     c d
b a       
2 1  3 4
```

(Throwing an exception on the line where `bar` is created is also possibly acceptable, although confusing.)

#### Produced Output

```
     c   d
b a       
2 1  3 NaN
```

#### Problem description

When joining with a data frame with a MultiIndex with a single level in it, left joins seem to not work properly. This behavior is confusing (why would a MultiIndex with a single index not work?), and is inconsistent with similar code which produces expected output.

Possibly related issues: #34292, #29252, #34357

#### Examples of similar code with expected behavior

Example 1:

<details>

```python
import pandas as pd

foo = pd.DataFrame(data={'c': 3}, index=pd.MultiIndex.from_tuples([(1, 2)], names=('a', 'b')))
bar = pd.DataFrame(data={'d': 4}, index=pd.Index([2], name='b'))
joined = foo.join(bar, how='left')
print(joined)
```
produces
```
     c  d
a b      
1 2  3  4
```

</details>

Example 2:

<details>

```python
import pandas as pd

foo = pd.DataFrame(data={'c': 3}, index=pd.MultiIndex.from_tuples([(1, 2, 99)], names=('a', 'b', 'x')))
bar = pd.DataFrame(data={'d': 4}, index=pd.MultiIndex.from_tuples([(2, 99)], names=('b', 'x')))
joined = foo.join(bar, how='left')
print(joined)
```
produces
```
        c  d
b x  a      
2 99 1  3  4
```

</details>

#### Output of ``pd.show_versions()``

<details>

```
INSTALLED VERSIONS
------------------
commit           : db08276bc116c438d3fdee492026f8223584c477
python           : 3.8.6.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.18362
machine          : AMD64
processor        : Intel64 Family 6 Model 142 Stepping 12, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : English_United States.1252

pandas           : 1.1.3
numpy            : 1.19.2
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.1.1
setuptools       : 50.3.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None
```

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #36909
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

``droplevel`` for ``MultiIndex`` returns an ``Index`` when the resulting ``MultiIndex`` had only one level. But if the input had only one level and no level should be dropped, the return was a ``MultiIndex``. This did not seem consistent, so I changed it, that in this case an Index would be returned too. If this is not the desired behavior, we could fix the ``join`` problems after calling this functions.

If this is not desired, we should add a note to the docstring, that the return for ``droplevel`` is a MultiIndex, if no level is dropped.

### CODE REVIEW

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -474,6 +474,7 @@ Reshaping
 - Bug in func :meth:`crosstab` when using multiple columns with ``margins=True`` and ``normalize=True`` (:issue:`35144`)
 - Bug in :meth:`DataFrame.agg` with ``func={'name':<FUNC>}`` incorrectly raising ``TypeError`` when ``DataFrame.columns==['Name']`` (:issue:`36212`)
 - Bug in :meth:`Series.transform` would give incorrect results or raise when the argument ``func`` was dictionary (:issue:`35811`)
+- Bug in :func:`join` over :class:`MultiIndex` returned wrong result, when one of both indexes had only one level (:issue:`36909`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:477]: move to 1.3

**Review by jreback — CHANGES_REQUESTED** [review]
@phofl if you can merge master will have a look

**Comment by phofl** [comment]
Done

File: pandas/tests/reshape/merge/test_join.py [hunk 2]
```diff
@@ -815,3 +815,20 @@ def test_join_cross(input_col, output_cols):
     result = left.join(right, how="cross", lsuffix="_x", rsuffix="_y")
     expected = DataFrame({output_cols[0]: [1, 1, 3, 3], output_cols[1]: [3, 4, 3, 4]})
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("how", ["left", "right", "inner", "outer"])
```
Comment (jreback) [inline pandas/tests/reshape/merge/test_join.py:820]: don't we have a fixture for this?

File: pandas/core/indexes/base.py [hunk 3]
```diff
@@ -1672,7 +1672,7 @@ def _drop_level_numbers(self, levnums: List[int]):
         Drop MultiIndex levels by level _number_, not name.
         """
 
-        if not levnums:
+        if not levnums and (len(self) > 1 or not isinstance(self, ABCMultiIndex)):
```
Comment (jreback) [inline pandas/core/indexes/base.py:1675]: this len(self) > 1 is a very odd check, what is the intent?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/reshape/merge/test_join.py (hunk 2 above)
Comment (phofl) [reply pandas/tests/reshape/merge/test_join.py:820]: Oh, yes. thx

File: pandas/core/indexes/base.py (hunk 3 above)
Comment (phofl) [reply pandas/core/indexes/base.py]: Seems to be no longer necessary. Code changed quite a bit since I implemented this. Removed it.

**Review by jreback — APPROVED** [review]

**Comment by phofl** [comment]
@jreback green

**Comment by jreback** [comment]
thanks

**Pull request merged**